### ISSUE TITLE
Gapless playback

### ISSUE THREAD

**Issue by OxygenCobalt** [OxygenCobalt/Auxio#35]
Self-explanatory. Would benefit listening experience for a lot of people [including myself].

The main blocker for this is the queue system. ExoPlayer automatically attempts gapless playback if you have a single queue, but my two-queue system makes it really hard for me to leverage this. The solution is to move Auxio to a single-queue system, but I'm not sure if I want to do that,

**Comment by OxygenCobalt** [comment]
Sadly, this feature can't be implemented. This is for a few reasons:

1. It completely destroys the playback system's coherency, forcing us to have a perfect mirror of the queue at all times and also requiring the player to borg much of the functionality in `PlaybackStateManager`. This makes things incredibly hard to maintain and much more bug-prone.
2. Handling reshuffling events is a mess. For some insane reason, ExoPlayer's queue functionality makes it as so every time the queue is changed, *the position in playback and the queue is reset to 0.* This requires me to do a bunch of programming tricks to preserve the player position on a re-shuffle event, but that results in the playback buffering every time the queue is reshuffled. I could fix this by using ExoPlayer's shuffle functionality, but this makes problem 1 even worse.
3. I can't do what Phonograph does with `MultiPlayer` and chain ExoPlayer's together, as thats incredibly resource-inefficient. I would have to use the queue system, but again, that is a nightmarish mess.

Basically, implementing gapless playback requires ripping out pretty much the entirety of Auxio's playback system and using android components instead. This avoided these components for a reason. They are extremely hard to customize and create far too many behavior quirks to preserve Auxio's sensible nature. I would rather take minor gaps of silence between files than deal with this.
